src/extension/sdk/dev_tools/manager.ts

```diff
@@ -57,8 +58,9 @@ export class DevToolsManager implements vs.Disposable {
 	private async handleEagerActivationAndStartup(workspaceContext: DartWorkspaceContext) {
 		if (workspaceContext.config?.startDevToolsServerEagerly) {
 			try {
-				if (workspaceContext.config?.startDevToolsServerEagerly)
-					await this.spawnIfRequired(true);
+				if (workspaceContext.config?.startDevToolsServerEagerly) {
+					await this.spawnIfRequired(workspaceContext.config?.startDevToolsFromDaemon ? new FlutterDaemon(new EmittingLogger(), workspaceContext as FlutterWorkspaceContext, this.flutterCapabilities) : undefined, true);
```

Comment:
Will this spawn a second daemon process? If so, I think we should instead pass the existing daemon in (constructed in `extension.ts` - the code that creates it might need moving to be earlier in the activation, but I don't think that should cause any problems).